### PR TITLE
Handle iconic spells using the `CrucibleSpellAction` subclass

### LINKED ISSUE #477: Add a property/method/getter so that iconic spell actions automatically scale based on their parent talent's rune & gesture prerequisites

### LINKED ISSUE #667: Migrate iconic spells to use the `CrucibleSpellAction` class instead of the `CrucibleAction` class to consolidate all logic pertaining to spellcraft under a specialized class dealing with spells.

### PR DESCRIPTION
**Pull request by aaclayton**
Closes https://github.com/foundryvtt/crucible/issues/667
Closes https://github.com/foundryvtt/crucible/issues/477

### CODE REVIEW

File: module/models/item-spell.mjs [hunk 1]
```diff
@@ -51,9 +52,11 @@ export default class CrucibleSpellItem extends foundry.abstract.TypeDataModel {
    * @inheritDoc
    */
   _initializeSource(source, options) {
-    super._initializeSource(source, options);
+    source = super._initializeSource(source, options);
     for ( const action of source.actions ) {
-      action.tags.unshift("spell", "iconicSpell");
+      action.rune = source.runes[0]; // TODO eventually support multi-component spell actions?
+      action.gesture = source.gestures[0];
+      action.inflection = source.inflections[0];
```
Comment (roth-michael) [inline module/models/item-spell.mjs]:
```suggestion
      action.rune = source.runes[0] ?? ""; // TODO eventually support multi-component spell actions?
      action.gesture = source.gestures[0] ?? "";
      action.inflection = source.inflections[0] ?? "";
```
Since we don't enforce that an iconic spell item _must_ have one of each (and we've now allowed `""` for these schema fields), should `?? ""` each of these I think. That, or enforce the requirement of components.

**Review by roth-michael — CHANGES_REQUESTED** [review]
A few edge cases, a few non-edge cases. Did some creation of iconic spells to test, and some casting of composed and iconic, as well as counterspelling.
